be/src/olap/delta_writer.cpp

```diff
@@ -359,6 +361,17 @@ Status DeltaWriter::close_wait(const PSlaveTabletNodes& slave_tablet_nodes,
         return res;
     }
     if (_tablet->enable_unique_key_merge_on_write()) {
+        auto beta_rowset = reinterpret_cast<BetaRowset*>(_cur_rowset.get());
+        std::vector<segment_v2::SegmentSharedPtr> segments;
+        RETURN_IF_ERROR(beta_rowset->load_segments(&segments));
+        std::shared_lock meta_rlock(_tablet->get_header_lock());
```

Comment:
Is it necessary to hold the header lock?

Reply:
fixed